### PR TITLE
docs(PPDSC-2381): Tag documentation

### PR DESCRIPTION
**Pull request by RashikaNewsUK**
PPDSC-2381

**What**

1.Tag documentation UI.

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [X] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [X] The screen reader reads and flows through the elements as expected.
- [x] There are no new errors in the browser console coming from this PR.
- [x] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [x] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2381-tag-documentation
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2381-tag-documentation/storybook

File: cypress/site/functional/section-navigation.spec.js [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 describe('section navigation', () => {
   it('should contain the correct section navigation links', () => {
-    cy.mockConsentAndVisit('/components/tag');
+    cy.mockConsentAndVisit('/components/text-input');
```
Comment (mstuartf) [inline cypress/site/functional/section-navigation.spec.js:5]: Why does this need to be changed?

File: site/pages/components/tag.tsx [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+    name: 'size',
+    type: ['small', 'medium', 'large'],
+    default: 'medium',
+    description: 'Defines the size of the undefined tag.',
+  },
+  {
+    name: 'disabled',
+    type: 'boolean',
+    default: 'false',
+    description: 'If true, renders the undefined tag in a disabled state.',
```
Comment (mstuartf) [inline site/pages/components/tag.tsx:54]: What does `undefined` mean in this context? Might be a question for Mike M.

File: cypress/site/functional/section-navigation.spec.js (hunk 1 above)
Comment (Srinivasan-Ramasamy) [reply cypress/site/functional/section-navigation.spec.js:5]: Hi **@mstuartf**, Section navigation spec has cypress test scenario to check links in .mdx file. Actually part of current tag documentation changes we've removed the tag.mdx file. So we've updated the spec js with text-input.mdx file name to pass the build.

File: site/pages/components/tag.tsx (hunk 2 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Hi @mstuartf,
I have removed the word "**undefined**" from the API section. Could you please take look and confirm.

Thanks,
Rashika.

File: site/components/illustrations/components/tag/behaviours/fixedandfullwidth.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {Svg} from '../../../svg';
```
Comment (Vanals) [inline site/components/illustrations/components/tag/behaviours/fixedandfullwidth.tsx:2]: should this file be using kebab case!? If came like that probably should have been renamed correctly in figma in first place :)

Comment (RashikaNewsUK) [reply site/components/illustrations/components/tag/behaviours/fixedandfullwidth.tsx]: Hi @Vanals,

This is my mistake,while working on this doc I have modified the file name without any space. I am not aware about that.Apologize for that. I have corrected the file name according to the figma and will also correct on upcoming story as well.

Thanks,
Rashika.

File: site/pages/components/tag.tsx [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+  },
+];
+
+const commonOverridesRows = [
+  {
+    attribute: 'tag.size',
+    type: 'tagSize',
+    default: 'medium',
+    description: 'If provided, overrides the size of the tag.',
+  },
```
Comment (mutebg) [inline site/pages/components/tag.tsx:75]: this does not exist, can be removed

File: site/pages/components/tag.tsx [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+    type: 'MQ<string>',
+    description: (
+      <>
+        Overrides the minHeight of the tag.
+        <br />
+        <br />
+        This can be a sizing token from the theme, or any CSS length value.
+      </>
+    ),
+  },
```
Comment (mutebg) [inline site/pages/components/tag.tsx:99]: You can delete this one, its a duplicate. Bellow you have minHeight one more time

File: site/pages/components/tag.tsx [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+      'small = utilityButton010',
+      'medium =  utilityButton020',
+      'large = utilityButton030',
+    ],
+    description: 'If provided, overrides the typographyPreset of the tag.',
+  },
+  {
+    attribute: 'tag.transitionPreset',
+    type: 'MQ<string>',
+    default: ['backgroundColorChange', 'borderColorChange'],
```
Comment (mutebg) [inline site/pages/components/tag.tsx:145]:
```suggestion
    default: ['backgroundColorChange'],
```

File: site/pages/components/tag.tsx [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+    description: 'If provided, overrides the transitionPreset of the tag.',
+  },
+  {
+    attribute: 'tag.spaceInset',
```
Comment (mutebg) [inline site/pages/components/tag.tsx:149]:
```suggestion
    attribute: 'spaceInset',
```

File: site/pages/components/tag.tsx [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+      'medium = spaceInsetSquish020',
+      'large = spaceInsetSquish020',
+    ],
+    description:
+      "If provided, overrides the padding of the tag. Otherwise, use spaceInsetSquish010 or spaceInsetSquish020, depending on the 'size' prop.",
+  },
+  {
+    attribute: 'tag.spaceInline',
+    type: 'MQ<string>',
+    default: ['small = space010', 'medium = space010', 'large = space020'],
```
Comment (mutebg) [inline site/pages/components/tag.tsx:162]:
```suggestion
    default: ['small = space010', 'medium = space010', 'large = space010'],
```

File: site/pages/components/tag.tsx [hunk 9]
```diff
@@ -0,0 +1,564 @@
...
+    description:
+      'If provided, overrides the space between multiple children in the underlying stack.',
+  },
+  {
+    attribute: 'tag.icon.size',
```
Comment (mutebg) [inline site/pages/components/tag.tsx:167]:
```suggestion
    attribute: 'iconSize',
```

File: site/pages/components/tag.tsx [hunk 10]
```diff
@@ -0,0 +1,564 @@
...
+    type: 'MQ<string>',
+    default: ['small = space010', 'medium = space010', 'large = space020'],
```
Comment (mutebg) [inline site/pages/components/tag.tsx:169]:
```suggestion
    default: ['small = iconSize010', 'medium = iconSize010', 'large = iconSize020'],
```

File: site/pages/components/tag.tsx [hunk 11]
```diff
@@ -0,0 +1,564 @@
...
+            options: [
+              {
+                label: 'Default',
+                value: undefined,
+                isDefault: true,
+              },
+              {
+                label: 'Style Preset',
+                value: {
+                  stylePreset: 'tagCustom',
```
Comment (mutebg) [inline site/pages/components/tag.tsx:334]: tagCustom does not exist try with:
```suggestion
                  stylePreset: 'tagPrimaryInverse',
```

**Review by mutebg — CHANGES_REQUESTED** [review]

File: site/pages/components/tag.tsx [hunk 12]
```diff
@@ -0,0 +1,564 @@
...
+    type: 'boolean',
+    default: 'false',
+    description: 'If true, renders the tag in a disabled state.',
+  },
+  {
+    name: 'eventOriginator',
+    type: 'string',
+    default: 'tag',
+    description: 'Allows users to add event originator custom name.',
+  },
```
[the remark this answers is not in the document]
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: I have removed the event originator form the api section. -thanks!

File: site/pages/components/tag.tsx [hunk 13]
```diff
@@ -0,0 +1,564 @@
...
+  {
+    attribute: 'minHeight',
+    type: 'MQ<string>',
+    default: ['small = sizing050', 'medium = sizing060', 'large = sizing070'],
+    description:
+      'If provided, sets a minimum width to the tag. This can be a sizing token from the theme, or any CSS length value.',
+  },
+  {
+    attribute: 'tag.stylePreset',
```
[the remark this answers is not in the document]
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: I have modified all the attributes name which mentioned here. -thanks!

File: site/pages/components/tag.tsx (hunk 5 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: I have removed this one as well. -thanks!

File: site/pages/components/tag.tsx [hunk 14]
```diff
@@ -0,0 +1,564 @@
...
+        <br />
+        <br />
+        This can be a sizing token from the theme, or any CSS length value.
+      </>
+    ),
+  },
+  {
+    attribute: 'minWidth',
+    type: 'MQ<string>',
+    default: ['small = sizing090', 'medium = sizing100', 'large = sizing110'],
```
[the remark this answers is not in the document]
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Done, I have removed the default. -thanks!

File: site/pages/components/tag.tsx (hunk 6 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Done. Removed the "borderColorChange" from the default. -thanks!

File: site/pages/components/tag.tsx (hunk 8 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Done. Modified the large value. -thanks!

File: site/pages/components/tag.tsx (hunk 10 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Done. Modified the default values. -thanks!

File: site/pages/components/tag.tsx (hunk 11 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Done. Modified the stylePreset value. -thanks!

File: site/pages/components/tag.tsx (hunk 4 above)
Comment (RashikaNewsUK) [reply site/pages/components/tag.tsx]: Done

**Review by mutebg — APPROVED** [review]

**Review by Srinivasan-Ramasamy — APPROVED** [review]

**Pull request merged**